src/oic/oauth2/message.py

```diff
@@ -431,6 +431,12 @@ def _add_value(self, skey, vtyp, key, val, _deser, null_allowed):
         else:
             if val is None:
                 self._dict[skey] = None
+            elif isinstance(val, bool):
+                if vtyp is bool:
```

Comment:
vtyp is not an instance of bool. It is the type bool so doing an isinstance test won't work.